repose-aggregator/core/repose-core/src/main/java/org/openrepose/powerfilter/PowerFilter.java

```diff
@@ -385,42 +387,25 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
         // Re-wrapping the request to reset the inputStream/Reader flag
         wrappedRequest = new HttpServletRequestWrapper((HttpServletRequest) request, bufferedInputStream);
 
-        // OpenTracing - set up scope placeholder.
-        Scope scope = null;
-
-        if (openTracingService.isPresent() && openTracingService.get().isEnabled()) {
-            LOG.trace("We enabled OpenTracing service.  Let's see if there are any passed-in spans");
-
-            SpanContext context = null;
-
-            try {
-                context = openTracingService.get().getGlobalTracer().extract(
-                    Format.Builtin.HTTP_HEADERS, new TracerExtractor(wrappedRequest));
-            } catch (RuntimeException re) {
-                LOG.error("Incoming tracer could not be parsed.  We're going to start a new root span here; even though" +
-                    "this is most likely part of a larger span.  Check out thrown exception for more details", re);
-            }
-
-            LOG.debug("Got the span context from request: {}", context);
-
-            Span activeSpan;
-
-            if (context == null)
-                activeSpan = openTracingService.get().getGlobalTracer()
-                    .buildSpan(String.format("%s %s", wrappedRequest.getMethod(),
-                        wrappedRequest.getRequestURI()))
-                    .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_CLIENT).start();
-            else
-                activeSpan = openTracingService.get().getGlobalTracer()
-                    .buildSpan(String.format("%s %s", wrappedRequest.getMethod(),
-                        wrappedRequest.getRequestURI()))
-                    .asChildOf(context)
-                    .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_CLIENT).start();
-
-            scope = openTracingService.get().getGlobalTracer().scopeManager().activate(activeSpan, false);
+        LOG.trace("Let's see if there are any OpenTracing spans passed-in");
+        SpanContext context = null;
+        try {
+            context = tracer.extract(Format.Builtin.HTTP_HEADERS, new TracerExtractor(wrappedRequest));
+        } catch (RuntimeException re) {
+            LOG.error("Incoming tracer could not be parsed.");
+            LOG.error("We're going to start a new root span here; even though this is most likely part of a larger span.");
+            LOG.error("Check out following exception for more details:", re);
```

Comment:
I suspect making three separate `error` log calls means that these may not show up in order in the log (since other messages may squeeze between them). I worry that might be confusing for the user.

On the other hand, I totally get wanting these messages on separate lines.

On the other other hand, you can't actually guarantee that these messages will be on separate lines since that is up to the log format, right? And speaking of the log format, logging these messages separately will likely produce a bunch of redundant data in the logs (things like the thread ID, timestamp, etc.).

My gut feeling is that these should be collapsed into a single call, but I'm going to ask @adrianjgeorge what he thinks, and if we do collapse these, how we want to write it so that it is both clear in the code and the log.

Reply:
These were originally one big concatenated `String` and it was just ugly. I agree that there will be some redundant data and the possibility of interspersed messages, but the messages would all be linear as far as a given trace. I'm definitely open for suggestions on how we want to handle this large amount of data.

Reply:
Something like below would give the same result as the original prototype, but is a bit more readable:
```java
            LOG.error(String.format("%s %s %s",
                "Incoming tracer could not be parsed.",
                "We're going to start a new root span here; even though this is most likely part of a larger span.",
                "Check out following exception for more details:"
            ), re);
```
Even the following is still consistent with other locations of logging exceptions/stacktraces and has the added benefit of only doing the formatting if necessary:
```java
            LOG.error("%s %s",
                "Incoming tracer could not be parsed.",
                "We're going to start a new root span here; even though this is most likely part of a larger span.");
            LOG.error("Check out following exception for more details:", re);
```